### PR TITLE
Add settings for app mv and bash command execution

### PR DESCRIPTION
**Pull request by JoeSSS**
In order to do advanced settings that we for example do for XING project, we need to be able to move apps and run some bash before every test run. This PR adds two new fields that allow to do that.

(!) This is the last PR of the series. After merging this and all that is underneath, we can release a new build that can be used externally as well as internally.

### CODE REVIEW

File: Launcher/Classes/Controllers/CommandsController.swift [hunk 1]
```diff
@@ -3,6 +3,7 @@ import CommandsCore
 
 class CommandsController {
     let applicationStateHandler = ApplicationStateHandler()
+    let plistOperations = PlistOperations()
```
Comment (q231950) [inline Launcher/Classes/Controllers/CommandsController.swift]: Maybe we should rename PlistOperations to SettingsController/PlistHandler/PlistController? It has nothing to do with NSOperations. What do you think?

File: Launcher/Storyboards/Base.lproj/Main.storyboard [hunk 2]
```diff
@@ -1152,22 +1152,11 @@
...
-                            <button verticalHuggingPriority="750" fixedFrame="YES" translatesAutoresizingMaskIntoConstraints="NO" id="jPd-p7-nqK">
-                                <rect key="frame" x="556" y="17" width="136" height="28"/>
-                                <autoresizingMask key="autoresizingMask" flexibleMaxX="YES" flexibleMinY="YES"/>
-                                <buttonCell key="cell" type="push" title="Close" bezelStyle="rounded" alignment="center" controlSize="small" borderStyle="border" imageScaling="proportionallyDown" inset="2" id="It8-AQ-Odb">
-                                    <behavior key="behavior" pushIn="YES" lightByBackground="YES" lightByGray="YES"/>
-                                    <font key="font" metaFont="smallSystem"/>
-                                </buttonCell>
-                                <connections>
-                                    <action selector="clickCloseButton:" target="Nd7-tF-bGG" id="Tkb-Zm-wTR"/>
-                                </connections>
```
Comment (q231950) [inline Launcher/Storyboards/Base.lproj/Main.storyboard]: I am on my mobile device and can’t see the close button being added anywhere - can you help me find it? 😅

File: Launcher/Storyboards/Base.lproj/Main.storyboard [hunk 3]
```diff
@@ -1397,10 +1437,34 @@
...
+                            <button verticalHuggingPriority="750" fixedFrame="YES" translatesAutoresizingMaskIntoConstraints="NO" id="jPd-p7-nqK">
+                                <rect key="frame" x="556" y="14" width="136" height="28"/>
+                                <autoresizingMask key="autoresizingMask" flexibleMaxX="YES" flexibleMinY="YES"/>
+                                <buttonCell key="cell" type="push" title="Close" bezelStyle="rounded" alignment="center" controlSize="small" borderStyle="border" imageScaling="proportionallyDown" inset="2" id="It8-AQ-Odb">
+                                    <behavior key="behavior" pushIn="YES" lightByBackground="YES" lightByGray="YES"/>
+                                    <font key="font" metaFont="smallSystem"/>
+                                </buttonCell>
+                                <connections>
+                                    <action selector="clickCloseButton:" target="Nd7-tF-bGG" id="Tkb-Zm-wTR"/>
+                                </connections>
```
Comment (q231950) [inline Launcher/Storyboards/Base.lproj/Main.storyboard]: I found it!

File: Launcher/Storyboards/Base.lproj/Main.storyboard [hunk 4]
```diff
@@ -1853,6 +1917,54 @@ Please choose one of the supported devices from "Test Runner" menu and click "St
...
+            <objects>
+                <viewController id="Fzd-fW-W4C" sceneMemberID="viewController">
+                    <view key="view" id="JPz-gQ-5tw">
+                        <rect key="frame" x="0.0" y="0.0" width="193" height="44"/>
+                        <autoresizingMask key="autoresizingMask"/>
+                        <subviews>
+                            <textField verticalHuggingPriority="750" horizontalCompressionResistancePriority="250" fixedFrame="YES" translatesAutoresizingMaskIntoConstraints="NO" id="4WA-AF-Tik">
+                                <rect key="frame" x="8" y="-6" width="179" height="46"/>
+                                <autoresizingMask key="autoresizingMask" flexibleMaxX="YES" flexibleMinY="YES"/>
+                                <textFieldCell key="cell" controlSize="mini" sendsActionOnEndEditing="YES" title="Identify directory to copy the build after downloading." id="UxM-nB-nnN">
```
Comment (q231950) [inline Launcher/Storyboards/Base.lproj/Main.storyboard]: ~~Would it make sense to call it: Location of the application under test. ? Not everyone might download the app like we do~~

File: Scripts/bash/BuildScript.command [hunk 5]
```diff
@@ -13,10 +13,12 @@ export ${1}
 export ${2}
 # Env variables (can be empty)
 ${6}
-# Device IP (can be empty)
+# Command to execute before run (user defined)
```
Comment (q231950) [inline Scripts/bash/BuildScript.command]: It’s probably also optional like device ip?

File: Launcher/Classes/Controllers/CommandsController.swift (hunk 1 above)
Comment (JoeSSS) [reply Launcher/Classes/Controllers/CommandsController.swift]: I choose `PlistHandler` :)

File: Launcher/Classes/ViewControllers/SettingsViewController.swift [hunk 6]
```diff
@@ -36,16 +38,24 @@ class SettingsViewController: NSViewController {
...
-        let linkDescriptionArray = plistOperations.readValues(forKey: Constants.Keys.linkInfo)
+        if let pathToBuild = plistHandler.readValues(forKey: Constants.Keys.pathToBuildInfo).first {
+            appPathField.stringValue = pathToBuild
+        }
+        
+        if let command = plistHandler.readValues(forKey: Constants.Keys.commandFieldInfo).first {
+            commandField.stringValue = command
+        }
+        
+        let linkArray = plistHandler.readKeys(forKey: Constants.Keys.linkInfo)
```
Comment (BasThomas) [inline Launcher/Classes/ViewControllers/SettingsViewController.swift]: `linkKeys` maybe?

File: Launcher/Classes/ViewControllers/SettingsViewController.swift [hunk 7]
```diff
@@ -36,16 +38,24 @@ class SettingsViewController: NSViewController {
...
+        let linkDescriptionArray = plistHandler.readValues(forKey: Constants.Keys.linkInfo)
```
Comment (BasThomas) [inline Launcher/Classes/ViewControllers/SettingsViewController.swift]: `linkValues`?

File: Launcher/Classes/ViewControllers/SettingsViewController.swift [hunk 8]
```diff
@@ -151,13 +165,13 @@ class SettingsViewController: NSViewController {
         }
     }
     
-    func fillDictionary(with value: String, for key: String, on fieldName: String) -> [String : Any] {
+    func fillDictionary(with value: String, for key: String) -> [String : Any] {
```
Comment (BasThomas) [inline Launcher/Classes/ViewControllers/SettingsViewController.swift]: Looking at this, I have no idea what this function is doing / trying to accomplish. Can you explain what it is you're expecting here?

File: Launcher/Classes/ViewControllers/SettingsViewController.swift [hunk 9]
```diff
@@ -151,13 +165,13 @@ class SettingsViewController: NSViewController {
...
         var data: [String: Any] = [key : []]
         data[key] = [:]
         
         var existingItems =  data[key] as? [[String: String]] ?? []
```
Comment (BasThomas) [inline Launcher/Classes/ViewControllers/SettingsViewController.swift]: Nit: double space after `=`.

File: Launcher/Classes/ViewControllers/SettingsViewController.swift (hunk 8 above)
Comment (JoeSSS) [reply Launcher/Classes/ViewControllers/SettingsViewController.swift]: It fills the `Dictionary` with 'some' values for 'some' key. Will rename the method

**Pull request merged**